openedx/features/badging/tests/test_views.py

```diff
@@ -72,6 +88,66 @@ def test_trophycase_with_some_earned_badges(self, mock_populate_trophycase):
         badging_views.trophycase(request)
         mock_populate_trophycase.assert_called_once_with(request.user, mock.ANY, [user_badge])
 
+    @ddt.data(STATUS_CODE_TRUE, STATUS_CODE_FALSE)
+    @mock.patch('openedx.features.badging.views.render_to_response')
+    @mock.patch('openedx.features.badging.views.add_posts_count_in_badges_list')
+    @mock.patch('openedx.features.badging.views.NodeBBClient')
+    @mock.patch('openedx.features.badging.views.get_badge_progress_request_data')
+    @mock.patch('openedx.features.badging.views.get_discussion_team_ids')
+    def test_trophycase_get_progress_api(self, api_status_code, mock_get_discussion_team_ids,
+                                         mock_get_badge_progress_request_data,
+                                         mock_nodebb_client,
+                                         mock_add_posts_count_in_badges_list,
+                                         mock_render_response):
+        """
+        Test nodeBB api client call arguments and its response. Api will node be called it is mocked so we are just
+        setting response of api and later check if that response is using or not.
+        :param api_status_code: Status code that will be used to test both positive and negative case
+        :param mock_get_discussion_team_ids: mock get_discussion_team_ids helper method
+        :param mock_get_badge_progress_request_data: mock get_badge_progress_request_data helper method
+        :param mock_nodebb_client: mock NodeBB Client class
+        :param mock_add_posts_count_in_badges_list: mock add_posts_count_in_badges_list helper method
+        :param mock_render_response: mock render_to_response method because we are having problems with our templates
+        :return: None
+        """
+        request = self.request_factory.get(reverse('trophycase'), {'json': True})
+        request.user = self.user
+
+        second_course = CourseFactory(org="test", number="456", run="1")
+
+        # Enroll user in both courses
+        CourseEnrollmentFactory(user=self.user, course_id=self.course.id, is_active=True)
+        CourseEnrollmentFactory(user=self.user, course_id=second_course.id, is_active=True)
+
+        DiscussionCommunityFactory(course_id=self.course.id, community_url='100/abc')
+        DiscussionCommunityFactory(course_id=second_course.id, community_url='101/abc')
+
+        courses_discussion_data = [
+            {str(self.course.id): {DISCUSSION_ID_KEY: 100}},
+            {str(second_course.id): {DISCUSSION_ID_KEY: 101}}
+        ]
+
+        badges, sample_api_request, sample_api_response = get_all_data_dicts(self, courses_discussion_data)
+
+        courses_discussion_data[0][str(self.course.id)][DISCUSSION_COUNT_KEY] = 3  # Could be any number
+        courses_discussion_data[1][str(second_course.id)][DISCUSSION_COUNT_KEY] = 2  # Could be any number
```

Comment:
Can we set count key directly in `courses_discussion_data` at line no 125, for both courses